File: bioinformatics-stronghold/revc.scala
// Problem ID: REVC
// Complementing a strand of DNA

import scala.io.Source

def reverseComplement(s: String): String = {
    var pairs = Map('A' -> 'T', 'T' -> 'A', 'C' -> 'G', 'G' -> 'C')
    s.trim.reverse.map(pairs(_))
}

val s = "AAAACCCGGT"
assert(reverseComplement(s) == "ACCGGGTTTT")

var file = Source.fromFile("data/rosalind_revc.txt").mkString
println(reverseComplement(file))

File: bioinformatics-stronghold/revc.py
"""
Problem ID: REVC
http://rosalind.info/problems/revc/

Complementing a Strand of DNA
"""

def reverse_complement(s):
    pairs = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C'}
    reversed_ = s.strip()[::-1]
    
    complement = ''
    for letter in reversed_:
        complement += pairs[letter]
    return complement


s = 'AAAACCCGGT'
assert(reverse_complement(s) == 'ACCGGGTTTT')


with open('data/rosalind_revc.txt', 'r') as f:
    print(reverse_complement(f.read()))